p2p/providers/peerstore_provider/rpc/provider.go

```diff
@@ -99,15 +107,15 @@ func (rpcPSP *rpcPeerstoreProvider) initRPCClient() {
 // options
 
 // WithP2PConfig allows to specify a custom P2P config
-func WithP2PConfig(p2pCfg *configs.P2PConfig) modules.ModuleOption {
-	return func(rabp modules.InitializableModule) {
-		rabp.(*rpcPeerstoreProvider).p2pCfg = p2pCfg
+func WithP2PConfig(p2pCfg *configs.P2PConfig) rpcPeerstoreProviderOption {
+	return func(rpcPSP *rpcPeerstoreProvider) {
+		rpcPSP.p2pCfg = p2pCfg
```

Comment:
I think this can actually be removed, it doesn't look like `rpcPeerstoreProvider#p2pCfg` is used anywhere.